### PR TITLE
Using placeholder rustdoc generation for current crate

### PR DESCRIPTION
**Pull request by tonowak**
This PR is stacked on https://github.com/obi1kenobi/cargo-semver-checks/pull/350.

It changes the logic of the tool. Instead of generating the rustdoc of the current crate directly, it uses the baseline's way of creating a placeholder project and running rustdoc on it.

### CODE REVIEW

**Comment by tonowak** [comment]
This PR should fix https://github.com/obi1kenobi/cargo-semver-checks/issues/167, but I don't have an Apple I could use to check it. @obi1kenobi can you test whether this PR fixes the issue?

File: src/manifest.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use anyhow::Context as _;
```
Comment (obi1kenobi) [inline src/manifest.rs:1]: Do we need the `as _` bit here?

**Review by obi1kenobi — COMMENTED** [review]
This looks good too, just the one nitpick.

**Comment by obi1kenobi** [comment]
> This PR should fix #167, but I don't have an Apple I could use to check it. @obi1kenobi can you test whether this PR fixes the issue?

I don't have an Apple device either. Maybe we can ask the original reporter of #167 to try it once we release this.

File: src/manifest.rs (hunk 1 above)
Comment (tonowak) [reply src/manifest.rs]: Done in 3b18ce47cc9e14eaa9e63281968e8dc9eaf3420b. I was basing this code on `manifest.rs` which also had `as _`, so I removed it there also.

**Review by obi1kenobi — APPROVED** [review]

File: src/manifest.rs (hunk 1 above)
Comment (obi1kenobi) [reply src/manifest.rs]: Ah I think I just realized what was going on here. `Context` is a trait, probably? And the `as _` means that the trait's methods are brought into scope without bringing the trait itself into scope.

**Pull request merged**